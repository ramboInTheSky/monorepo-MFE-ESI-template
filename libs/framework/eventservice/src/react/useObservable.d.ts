import { ESB } from "../esb";
export declare const useCommonObservable: (esb: ESB, callback: (data: any) => void) => void;
