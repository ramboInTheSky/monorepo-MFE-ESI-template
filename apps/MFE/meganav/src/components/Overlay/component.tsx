import {Modal} from "@mui/material"
import styled from "styled-components"

/*
 * Title: Using to style top !important
 * problem: we do not want to use !important but we want a modal and a modal is expected to cover the entire screen
 * in this case however, material UI probably assumes you'll have the entire screen covered and uses inline styles
 * to define this behaviour. But here the modal as it is the case of next, partially exposes the header area hence it
 * does not fully cover the entire screen.
 * Solution: for now we have to use !important
 * Refs:
 *    https://github.com/mui-org/material-ui/issues/16441
 *    https://github.com/mui-org/material-ui/issues/16442
 */

export const Container = styled(Modal)`
    && {
        transition: opacity 0.2s ease;
        animation-delay: 250ms;
        display: flex;
        justify-content: center;
        position: relative !important;
        height: 100vh;

        & > div:first-child {
            position: absolute;
            background: rgba(0, 0, 0, 0.9);
            z-index: 1;
        }
        & > [data-test="sentinelStart"] {
            display: none;
        }
        & > [data-test="sentinelEnd"] {
            display: none;
        }
    }
`
