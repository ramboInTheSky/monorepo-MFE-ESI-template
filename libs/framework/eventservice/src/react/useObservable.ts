import {useEffect} from "react"
import {ESB} from "../esb"

export const useCommonObservable = (esb: ESB, callback: (data: any) => void) => {
    return useEffect(() => {
        const {subscription} = esb.subscribe(callback)
        return () => {
            subscription.unsubscribe()
        }
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])
}
