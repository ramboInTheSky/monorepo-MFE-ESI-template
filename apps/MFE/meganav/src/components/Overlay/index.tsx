import React, {useEffect, useState} from "react"
import {Backdrop} from "@mui/material"
import {Container} from "./component"
import {OVERLAY_TIMEOUT, PRIMARY_NAV_ITEM_HOVER_DELAY} from "../../config/constants"

export type OverlayProps = {
    open: boolean
    children: JSX.Element
    handleClose: () => void
}
const innerOverlayDefaultProps = {
    disableRestoreFocus: true,
    disableAutoFocus: true,
    disableEnforceFocus: false,
    disablePortal: true,
    keepMounted: false,
    closeAfterTransition: true,
    disableScrollLock: true,
    disableEscapeKeyDown: false,
}
let timerId: any = 0
const Overlay = ({open, children, handleClose}: OverlayProps) => {
    const [delayedOpen, setOpen] = useState(false)
    useEffect(() => {
        timerId = setTimeout(() => setOpen(open), PRIMARY_NAV_ITEM_HOVER_DELAY)
        return () => {
            handleClose()
            clearTimeout(timerId)
        }
    }, [open, handleClose])

    return (
        <Container
            {...innerOverlayDefaultProps}
            data-testid="overlay"
            open={delayedOpen}
            BackdropComponent={Backdrop}
            BackdropProps={{timeout: OVERLAY_TIMEOUT}}
            onClose={handleClose}
        >
            <>{children}</>
        </Container>
    )
}

export default Overlay
