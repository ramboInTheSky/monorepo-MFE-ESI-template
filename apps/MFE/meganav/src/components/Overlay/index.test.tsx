import React from "react"
import {render, screen, waitForElement} from "@testing-library/react"
import Overlay, {OverlayProps} from "."

jest.useFakeTimers()
jest.mock("../../config/constants", () => ({
    OVERLAY_TIMEOUT: 30,
    PRIMARY_NAV_ITEM_HOVER_DELAY: 0,
}))
const Fake = () => <div> fake </div>

describe("Components - Overlay: ", () => {
    let props: Omit<OverlayProps, "children">
    beforeEach(() => {
        props = {
            open: false,
            handleClose: jest.fn(),
        }
    })
    it("should match the snapshot", () => {
        const {asFragment} = render(
            <Overlay {...props}>
                <Fake />
            </Overlay>,
        )
        expect(asFragment()).toMatchSnapshot()
    })
    it("Should be visible when open state is true", async () => {
        render(
            <Overlay {...props} open>
                <Fake />
            </Overlay>,
        )
        jest.advanceTimersByTime(300)
        const got = await waitForElement(() => screen.getByTestId("overlay"))
        expect(got).toBeVisible()
        expect(got).toContainHTML("<div> fake </div>")
    })
})
