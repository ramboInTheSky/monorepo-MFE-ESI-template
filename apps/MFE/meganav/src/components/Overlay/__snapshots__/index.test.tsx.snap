// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Overlay:  should match the snapshot 1`] = `<DocumentFragment />`;
